sdk/core/azure-core/src/main/java/com/azure/core/util/paging/ContinuablePagedFluxCore.java

```diff
@@ -156,23 +152,25 @@ public void subscribe(CoreSubscriber<? super T> coreSubscriber) {
      * @param state the state to be used across multiple Page Retriever Function calls
      * @param pageRetriever the Page Retriever Function
      * @param pageSize the preferred page size
-     * @param <C> the type of Continuation token
-     * @param <T> The type of items in a {@link ContinuablePage}
-     * @param <P> The {@link ContinuablePage} holding items of type {@code T}
      * @return a Flux of {@link ContinuablePage}
      */
-    private static <C, T, P extends ContinuablePage<C, T>> Flux<P> concatFluxOfPage(ContinuationState<C> state,
-        PageRetriever<C, P> pageRetriever, Integer pageSize) {
+    private Flux<P> retrievePages(ContinuationState<C> state, PageRetriever<C, P> pageRetriever, Integer pageSize) {
+        return retrievePage(state, pageRetriever, pageSize)
```

Comment:
I think it makes no difference in our case; i.e. storing the continuation-token before we gave the page to the user (i.e. in doOnNext immediately after call to "retrievePage") vs storing it after user consumes the page (inside expand).

Reply:
From what I remember being told it is better to have explicitly needed operations performed in `map` or `flatMap` as `doOnX` is meant for side affect operations such as logging or metrics. I don't see continuation token updating as a side affect but an explicit need.

Reply:
Yep, I think all we need to ensure is the link is tthere before retrieving the next page, which we do. lgtm.